Source/Scene/DracoLoader.js

```diff
@@ -0,0 +1,197 @@
+define([
+        '../Core/arraySlice',
+        '../Core/ComponentDatatype',
+        '../Core/defined',
+        '../Core/FeatureDetection',
+        '../Core/TaskProcessor',
+        '../Renderer/Buffer',
+        '../Renderer/BufferUsage',
+        '../ThirdParty/GltfPipeline/ForEach',
+        '../ThirdParty/when'
+    ], function(
+        arraySlice,
+        ComponentDatatype,
+        defined,
+        FeatureDetection,
+        TaskProcessor,
+        Buffer,
+        BufferUsage,
+        ForEach,
+        when) {
+    'use strict';
+
+    /**
+     * @private
+     */
+    function DracoLoader() {}
+
+    // Maximum concurrency to use when deocding draco models
+    DracoLoader._maxDecodingConcurrency = Math.max(FeatureDetection.hardwareConcurrency - 1, 1);
+
+    // Exposed for testing purposes
+    DracoLoader._decoderTaskProcessor = undefined;
+    DracoLoader._getDecoderTaskProcessor = function () {
+        if (!defined(DracoLoader._decoderTaskProcessor)) {
+            DracoLoader._decoderTaskProcessor = new TaskProcessor('decodeDraco', DracoLoader._maxDecodingConcurrency);
+        }
+
+        return DracoLoader._decoderTaskProcessor;
+    };
+
+    function hasExtension(model) {
+        return (defined(model.extensionsRequired.KHR_draco_mesh_compression)
+            || defined(model.extensionsUsed.KHR_draco_mesh_compression));
+    }
+
+    function addBufferToModelResources(model, buffer) {
+        var resourceBuffers = model._rendererResources.buffers;
+        var bufferViewId = Object.keys(resourceBuffers).length;
+        resourceBuffers[bufferViewId] = buffer;
+        model._geometryByteLength += buffer.sizeInBytes;
+
+        return bufferViewId;
+    }
+
+    function addNewVertexBuffer(typedArray, model, context) {
+        var vertexBuffer = Buffer.createVertexBuffer({
```

Comment:
It looks like this is the case.

@ggetz for this you could try integrating the draco loading with `createBuffers` which manages buffer creation from typed arrays.

Reply:
I've made changes in https://github.com/AnalyticalGraphicsInc/cesium/pull/6341 that ensure the buffers are being created during `Model.update`, does that cover this? Or would it be better to integrate it into `createBuffers`?

Reply:
I think it would be better, only for the sake of utilizing the `JobScheduler`. If it's easy to integrate I would try going for it.

However I did look at the approach in #6341 and it seems solid too.